src/sidecar/authStatusPolling.ts

```diff
@@ -60,12 +82,48 @@ export async function watchCCloudConnectionStatus(): Promise<void> {
     expiration: connection.status.authentication.requires_authentication_at,
     errors: connection.status.authentication.errors,
   });
-  if (connection.status.authentication.status !== "VALID_TOKEN") {
-    // INVALID_TOKEN or NO_TOKEN
-    logger.error("current CCloud connection has invalid or no token; invalidating auth session", {
-      status: connection.status.authentication.status,
-    });
+
+  if (connection.status.authentication.status !== "INVALID_TOKEN") {
+    // resolve any open progress notification if we see a non-`INVALID_TOKEN` status
+    nonInvalidTokenStatus.fire();
+    // and go back to polling every 10sec
+    pollCCloudConnectionAuth.useSlowFrequency();
+    // and set the flag back so the notification can open again if we see another `INVALID_TOKEN`
+    invalidTokenNotificationOpen = false;
+  }
+
+  if (["NO_TOKEN", "FAILED"].includes(connection.status.authentication.status)) {
+    // some unusable state that requires the user to reauthenticate
+    logger.error(
+      "current CCloud connection has no token or transitioned to a failed state; invalidating auth session",
+      {
+        status: connection.status.authentication.status,
+      },
+    );
     ccloudAuthSessionInvalidated.fire();
+  } else if (connection.status.authentication.status === "INVALID_TOKEN") {
+    // the sidecar is handling a transient error, so exit this polling iteration early and check
+    // the status again on the next iteration
+    logger.warn("current CCloud connection has an invalid token; waiting for updated status");
+    if (!invalidTokenNotificationOpen) {
+      invalidTokenNotificationOpen = true;
+      vscode.window.withProgress(
+        {
+          location: vscode.ProgressLocation.Notification,
+          title: "Attempting to reconnect to Confluent Cloud...",
```

Comment:
I don't this is actionable in any way… May be even slightly irritating if it happens to pop up when I don't even use the extension, but have it activated while doing other things